ow_faults_detection/include/ow_faults_detection/FaultDetector.h

```diff
@@ -91,6 +93,7 @@ class FaultDetector
   ros::Subscriber m_camera_raw_sub;
   ros::Subscriber m_power_soc_sub;
   ros::Subscriber m_power_temperature_sub;
+  ros::Subscriber arm_faults_internal_sub;
```

Comment:
Optional improvement: individual, named subscriber members are not necessary, since they are only created and never referenced again.  One could replace all these members with a container, e.g. `std::list<ros::Subscriber> m_subscribers;` and then just push the new subscribers into this list.  Easier to maintain.

Reply:
You are correct, but on the other hand I like being able to see exactly what topics a class subscribes to from its header.